src/pr-trigger.ts

```diff
@@ -5,158 +5,146 @@ import { process as computeActions } from "./compute-pr-actions";
 import { executePrActions } from "./execute-pr-actions";
 import { mergeCodeOwnersOnGreen } from "./side-effects/merge-codeowner-prs";
 import { HttpRequest, Context } from "@azure/functions";
-import { Webhooks, EventPayloads } from "@octokit/webhooks";
+import { createEventHandler, EmitterWebhookEvent, verify } from "@octokit/webhooks";
 
 const prHandlers: { [prNumber: number]: () => void } = {};
 
 export async function httpTrigger(context: Context, req: HttpRequest) {
-
-    if (!(process.env["BOT_AUTH_TOKEN"] || process.env["AUTH_TOKEN"])) {
-        throw new Error("Set either BOT_AUTH_TOKEN or AUTH_TOKEN to a valid auth token");
-    }
-
     context.log(`[${process.version}] HTTP trigger function received a request.`);
 
-    const event = req.headers["x-github-event"];
-    if (!event) throw new Error(`Did not get a x-github-event header`);
```

Comment:
Checked in `eventHandler.receive()`

Reply:
> Side question: is this documented? When I tried to grep the code, I found `getMissingHeaders`, which points to what looks like a broken/outdated url: https://developer.github.com/webhooks/#delivery-headers
> 
> (That anchor is not there, and I don't see "delivery" or "headers" mentioned on that page.)

Looks like it, I think this section is what it must have once referenced: https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#delivery-headers

I opened an PR to update that comment https://github.com/octokit/webhooks.js/pull/467

> Another side thing -- I saw this line there
> 
> ```
> debugWebhooks(`${eventName} event received (id: ${id})`);
> ```
> 
> which made me think that it might be useful to print it, since there were several times where I was trying to cross-reference the bot's log with the GH webhook logs by guessing.

Done, I revised this to pass `context.log` to `createEventHandler()`, so it should print it now. :heavy_check_mark: https://github.com/DefinitelyTyped/dt-mergebot/blob/d391d0494a2784ad257871a44669d3561337fa77/src/pr-trigger.ts#L42

Reply:
👍